neqo-common/src/datagram.rs

```diff
@@ -46,15 +37,43 @@ impl Datagram {
     }
 }
 
-impl Deref for Datagram {
-    type Target = Vec<u8>;
+impl<D: AsRef<[u8]>> Datagram<D> {
+    pub fn len(&self) -> usize {
+        self.d.as_ref().len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+}
+
+// TODO: Needed?
```

Comment:
It's harmless if not, though I think it might make some of our tests easier.

Reply:
Removed `TODO` and moved behing `#[cfg(test)]`. In case we ever need it in non-tests, we can remove the `cfg`.